### PR TITLE
Stage3

### PR DESCRIPTION
**Pull request by VeronikaVasilchikova**
Date: 2020/03/23

Here is my implementation of stage3 accordint to requirements
https://hackmd.io/@listopad/HkKv8MNlM

I'll be glad to get your feedback and comments :)

### CODE REVIEW

File: sources/views/activities.js [hunk 1]
```diff
@@ -131,6 +162,46 @@ export default class ActivitiesView extends JetView {
...
+						return webix.Date.equal(date.beginCurrentWeek, beginWeek) && value === "Open";
+					}
+					if (filter === "thismonth") {
+						return webix.Date.equal(date.beginCurrentMonth, beginMonth) && value === "Open";
+					}
+					return item;
+				}
+			},
+			{
+				getValue: node => node.getValue(),
```
Comment (HelgaListopad) [inline sources/views/activities.js]: This name of the parameter (`node`) is suitable if you handle an HTML element as a filter (a basic input, for example)
Here it's rather "view", or "control", or "input", or any reasonable name

File: sources/views/popupFormForSettings.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+		this.getRoot().getHead().setHTML(someAction);
+		this.$$("btn").setValue(someAction);
+	}
+
+	closeForm() {
+		this.form.clear();
+		this.getRoot().hide();
+	}
+
+	addOrEdit() {
```
Comment (HelgaListopad) [inline sources/views/popupFormForSettings.js]: Would be great to add validation here 
- required name of activity
- required icon (as you don't have "empty" property in suggest)

File: sources/views/popupFormForSettings.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+				this.form.setValues({
+					id: item.id,
+					Icon: "",
+					Value: item.Value
+				});
+			}
+		}
+		this.getRoot().show();
+		const someAction = id ? _("Edit") : _("Add");
+		this.getRoot().getHead().setHTML(someAction);
```
Comment (HelgaListopad) [inline sources/views/popupFormForSettings.js]: Also add a name of the instance you edit (activity type of status)

File: sources/views/popupFormForSettings.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+			width: 450,
+			move: true,
+			head: " ",
+			body: {
+				view: "form",
+				localId: "form",
+				elements: [
+					{
+						view: "text",
+						label: _(this.label),
```
Comment (HelgaListopad) [inline sources/views/popupFormForSettings.js]: Plural "types" or "statuses" doesn't seem right in this label
image

File: sources/views/activities.js (hunk 1 above)
Comment (VeronikaVasilchikova) [reply sources/views/activities.js]: I fixed it

File: sources/views/popupFormForSettings.js (hunk 2 above)
Comment (VeronikaVasilchikova) [reply sources/views/popupFormForSettings.js]: I fixed it

File: sources/views/popupFormForSettings.js (hunk 4 above)
Comment (VeronikaVasilchikova) [reply sources/views/popupFormForSettings.js]: I fixed it

File: sources/views/popupFormForSettings.js (hunk 3 above)
Comment (VeronikaVasilchikova) [reply sources/views/popupFormForSettings.js]: I fixed it

**Pull request merged**